consensus/dummy/dynamic_fees.go

```diff
@@ -37,22 +37,37 @@ var (
 
 // CalcExtraPrefix takes the previous header and the timestamp of its child
 // block and calculates the expected extra prefix for the child block.
-//
-// CalcExtraPrefix should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
 func CalcExtraPrefix(
 	config *params.ChainConfig,
 	parent *types.Header,
 	timestamp uint64,
 ) ([]byte, error) {
-	window, err := calcFeeWindow(config, parent, timestamp)
-	return window.Bytes(), err
+	switch {
+	case config.IsApricotPhase3(timestamp):
+		window, err := calcFeeWindow(config, parent, timestamp)
+		return window.Bytes(), err
+	default:
+		// Prior to AP3 there was no expected extra prefix.
+		return nil, nil
+	}
 }
 
 // CalcBaseFee takes the previous header and the timestamp of its child block
 // and calculates the expected base fee for the child block.
 //
-// CalcBaseFee should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
+// Prior to AP3, the returned base fee will be nil.
```

Comment:
I think this approach is better since it enforces this without having the consumer (dev) of this function pay attention to this comment.